crates/noirc_evaluator/src/brillig/brillig_ir.rs

```diff
@@ -60,12 +85,32 @@ impl BrilligContext {
 
     /// Allocates an array of size `size` and stores the pointer to the array
     /// in `pointer_register`
-    pub(crate) fn allocate_array(&mut self, pointer_register: RegisterIndex, size: u32) {
-        let allocation = self.memory.allocate(size as usize);
+    pub(crate) fn allocate_fixed_length_array(
+        &mut self,
+        pointer_register: RegisterIndex,
+        size: usize,
+    ) {
+        let size_register = self.make_constant(size.into());
+        self.allocate_array_instruction(pointer_register, size_register);
+    }
 
-        self.push_opcode(BrilligOpcode::Const {
+    /// Allocates an array of size contained in size_register and stores the
+    /// pointer to the array in `pointer_register`
+    pub(crate) fn allocate_array_instruction(
+        &mut self,
+        pointer_register: RegisterIndex,
+        size_register: RegisterIndex,
+    ) {
+        self.push_opcode(BrilligOpcode::Mov {
             destination: pointer_register,
-            value: Value::from(allocation),
+            source: ReservedRegisters::alloc(),
+        });
+        self.push_opcode(BrilligOpcode::BinaryIntOp {
+            destination: ReservedRegisters::alloc(),
+            op: BinaryIntOp::Add,
+            bit_size: BRILLIG_INTEGER_ARITHMETIC_BIT_SIZE,
```

Comment:
I rather use a standard 64 bit in case these semantics ever get optimized and I dont think we lose anything, just my 2 cents